File: factory_pattern/python/pizza_store/simple_pizza_factory.py
from pizza import Pizza, CheesePizza, PepperoniPizza, VeggiePizza, ClamPizza


class SimplePizzaFactory:

    @staticmethod
    def order_pizza(pizza_type: str) -> Pizza:
        pizza: Pizza = None

        if pizza_type == 'cheese':
            pizza = CheesePizza()
        elif pizza_type == 'pepperoni':
            pizza = PepperoniPizza()
        elif pizza_type == 'veggie':
            pizza = VeggiePizza()
        elif pizza_type == 'clam':
            pizza = ClamPizza()

        return pizza

File: factory_pattern/java/pizza_store/simple_pizza_factory/SimplePizzaFactory.java
package pizza_store.simple_pizza_factory;

import pizza_store.pizza.Pizza;


public class SimplePizzaFactory {
    public Pizza orderPizza(String type) {
        Pizza pizza = null;

        if (type.equals("cheese")) {
            pizza = new CheesePizza();
        } else if (type.equals("pepperoni")) {
            pizza = new PepperoniPizza();
        } else if (type.equals("veggie")) {
            pizza = new VeggiePizza();
        } else if (type.equals("clam")) {
            pizza = new ClamPizza();
        }

        return pizza;
    }
}